src/components/Select2.vue

```diff
@@ -72,63 +73,97 @@ export default {
     };
   },
   methods: {
-    toggleSelect: function(init) {
+    toggleSelect(init) {
       if (!init) {
         this.$refs.select.classList.toggle('active');
 
         Fliplet.Widget.autosize();
       }
     },
-    setOption: function(value, init) {
+    setOption(value, init) {
       this.setedOption = this.getSelectedOptionLabel(value);
-      this.$emit('onSelect', value);
+      this.$emit('update:selectedOption', value);
       this.toggleSelect(init);
     },
-    getSelectedOptionValue: function(value) {
+    getSelectedOptionValue(value) {
       if (!this.optionValueKey) {
         return value;
       }
 
       return value && value[this.optionValueKey];
     },
-    getSelectedOptionLabel: function(value) {
+    getSelectedOptionLabel(value) {
       if (!this.optionLabelKey) {
         return value;
       }
 
       return value && value[this.optionLabelKey];
     },
-    search: function() {
+    deepCopy(inObject) {
+      let outObject;
+      let value;
+      let key;
+
+      if (typeof inObject !== 'object' || inObject === null) {
+        return inObject; // Return the value if inObject is not an object
+      }
+
+      // Create an array or object to hold the values
+      outObject = Array.isArray(inObject) ? [] : {};
+
+      // eslint-disable-next-line guard-for-in
+      for (key in inObject) {
+        value = inObject[key];
+
+        // Recursively (deep) copy for nested objects, including arrays
+        outObject[key] = this.deepCopy(value);
+      }
+
+      return outObject;
+    },
+    search() {
+      // Using JSON.parse(JSON.stringify()) to make rip inner array bindings
+      const optionsCopy = this.deepCopy(this.options);
```

Comment:
Does this not work?

And does it need to be a deep clone instead of just a clone?

```suggestion
      const optionsCopy = _.deepClone(this.options);
```

Reply:
@tonytlwu Yes, we need to use deep clone here because we may get an array with nested options and we didn't want to mutate original options when we do a search.

But we will need to add lodash library to use `_.deepClone` and `this.deepCopy` works fine.
Should we add a library to use only one method from it?

Reply:
Use `_.deepClone` because we don't need to maintain, debug and review the code from lodash.